doc/elements-confidential-transactions.md

```diff
@@ -0,0 +1,480 @@
+
+A surjection proof is a ring signature over the differences between
+input and output commitments. The space required for a surjection
+proof in Elements is proportional to the number of transaction inputs.
+Each input adds approximately 40 bytes to the surjection proof.  In
+the Elements implementation, a maximum of 256 inputs is supported. See
+[2] for more information on serialization size, and [4] for more
+information on ring signatures.
+
+
+## Confidential Addresses
+
+A confidential address combines a segwit address and a public blinding
+key into a single checksummed string. This address format is called
+"blech32" and is based on the "bech32" format that was introduced for
+segwit. Liquid production addresses use the prefix "lq1". Liquid
+regtest (elementsregtest) addresses use the prefix "el1".   
+
+By default, the Elements RPC `getnewaddress` will return a
+confidential address. A non-confidential segwit address and a public
+blinding key may be combined with the RPC `createblindedaddress` to
+create a confidential address. 
+
+See the python script [../test/functional/test_framework/liquid_addr.py](../test/functional/test_framework/liquid_addr.py)
+for a reference implementation of blech32 addresses.
+
+## Workflow Considerations
+
+The steps for manually creating a confidential transaction using
+Elements RPCs are as follows: 
+
+1. `createrawtransaction` – adds inputs and outputs to an empty
+transaction. Any outputs using confidential addresses will be blinded.
+2. `fundrawtransaction` – adds inputs to pay for the transaction, and
+an output for change.
+3. `blindrawtransaction` - blinds the outputs of a transaction, using
+inputs from the wallet.
+4. `rawblindrawtransaction` - can be used to blind
+a transaction whose inputs are not from the local wallet; input
+blinding factors must be provided.
+5. `signrawtransaction` - signs inputs using the wallet. At this
+point, the transaction may be broadcasted, using `sendrawtransaction`.
+
+In the construction of a Pedersen Commitment, random blinding factors
+are used in every transaction output value commitment except the
+last. The last blinding factor is chosen so that the blinding factors
+(and therefore the commitments) sum to zero. As a consequence, when
+blinding a transaction, there must be at least one output available to
+"balance the blinding factors". The Elements RPCs
+`blindrawtransaction` and `blindrawtransaction` may add an additional
+zero-valued output.
+
+## Asset Issuances and Reissuances
+
+An asset issuance creates a non-zero amount of a new asset, and zero
+or more reissuance  token that may be used to create more of the same
+asset at a later time. Reissuance tokens are also called "inflation
+keys".  
+
+In Elements, there are four types of transaction inputs:
+
+1. "typical" inputs that spend UTXOs
+2. coinbase inputs
+3. peg-ins
+4. asset issuances/reissuances.  
+
+An asset issuance input defines the id of a new asset some non-zero amount
```

Comment:
```suggestion
An asset issuance input defines the ID of a new asset, some non-zero amount
```

Reply:
It appears as though the comma was not added. Was this intentional? I feel as through there should be a comma after "asset".

Reply:
fixed